crates/core/src/live_socket/tests/streaming.rs

```diff
@@ -1,7 +1,42 @@
+use std::sync::{Arc, Mutex};
+
+use tokio::sync::oneshot::{self, *};
+
 use super::*;
+use crate::dom::{ChangeType, DocumentChangeHandler, NodeData, NodeRef};
 
+// As of this commit the server sends a
+// stream even every 10_000 ms
+// This sampling interval should catch one
+const MAX_TRIES: u64 = 120;
+const MS_DELAY: u64 = 100;
+
+struct Inspector {
+    tx: std::sync::Mutex<Option<Sender<()>>>,
+}
+
+impl DocumentChangeHandler for Inspector {
+    fn handle(
+        &self,
+        _change_type: ChangeType,
+        _node_ref: Arc<NodeRef>,
+        _node_data: NodeData,
+        _parent: Option<Arc<NodeRef>>,
+    ) {
+        let tx = self
+            .tx
+            .lock()
+            .expect("lock poison")
+            .take()
+            .expect("Channel was None.");
+
+        tx.send(()).expect("Message Never Received.");
+    }
+}
+
+// Tests that streaming connects, and succeeds at parsing at least one delta.
 #[tokio::test]
-async fn streaming_connect() {
+async fn streaming_connect() -> Result<(), String> {
```

Comment:
I know that `Result` can be the return type on a test and it'd be nice to use the `?` in a test. Though it doesn't look like you're using the `?` operator in this test. What do you think about just using `panic!("Merging panicked")` (panic may not be the correct choice) in place of the `return Err(format!("Merging Panicked"));`. I guess my think is that `.expect("error thingy")` is nice in that it's very specific on the line and ideally the error (if the expect param is correct).

Reply:
fair enough. That is more sanitary.